src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/modelchecking/ModelCheckingStrategy.kt

```diff
@@ -515,6 +417,104 @@ internal class ModelCheckingStrategy(
     }
 }
 
+private fun collectExceptionsOrEmpty(failure: LincheckFailure): Map<Throwable, ExceptionNumberAndStacktrace> {
```

Comment:
Please do not move these method bodies -- it makes harder to track the changes

Reply:
I need to put it away from the `ModelCheckingStrategy` class, because it is called from `constructTraceForPlugin`, which is also moved away from that class. I put them in companion class instead of global scope

Reply:
Can we revert all this "moving around" changes? Let's refactor model checker / plugin interaction in a separate PR?

Reply:
Still, should we do this in a separate PR?
I think those methods shouldn't be in the `ModelCheckingStrategy` class in the first place. 
We can move them to `IdeaPlugin.kt` and refactor accordingly.

Reply:
Agree with @eupp, let's do the refactoring in a separate PR, please

Reply:
Here is the PR: https://github.com/JetBrains/lincheck/pull/465